File: TuduManayer/java/TuduManager/domain/src/test/java/net/seymourpoler/tudumanager/domain/todo/search/SearchTodoServiceShould.java
package net.seymourpoler.tudumanager.domain.todo.search;

import net.seymourpoler.tudumanager.domain.todo.search.models.Todo;
import org.junit.Before;
import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class SearchTodoServiceShould {

    private ISearchTodoRepository repository;

    private ISearchTodoService service;

    @Before
    public void setUp(){
        repository = mock(ISearchTodoRepository.class);
        service = new SearchTodoService(repository);
    }

    @Test
    public void
    accept_null_search_text(){
        var todos = service.search(null);

        assertThat(todos).isEqualTo(List.of());
    }

    @Test
    public void
    accept_string_empty_search_text(){
        var todos = service.search("");

        assertThat(todos).isEqualTo(List.of());
    }

    @Test
    public void
    accept_white_space_search_text(){
        var todos = service.search("    ");

        assertThat(todos).isEqualTo(List.of());
    }

    @Test
    public void
    return_todos_by_search_text(){
        final String searchText = "search-text";
        var todos = List.of(new Todo(1, "title"));
        when(repository.search(searchText)).thenReturn(todos);

        var result = service.search(searchText);

        assertThat(result).isEqualTo(todos);
    }
}

File: TuduManayer/dotnet/react.mvc-core.postgres.dapper/TuduManager/test/TuduManayer.Domain.Test/Todo/Search/SearchTodoServiceShould.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Moq;
using Shouldly;
using TuduManayer.Domain.Todo.Search;
using Xunit;

namespace TuduManayer.Domain.Test.Todo.Search
{
    public class SearchTodoServiceShould
    {
        private Mock<ISearchTodoRepository> repository;
        private ISearchTodoService service;

        public SearchTodoServiceShould()
        {
            repository = new Mock<ISearchTodoRepository>();
            service = new SearchTodoService(repository.Object);
        }

        [Fact]
        public void return_todos()
        {
            const string searchText = "search-text";
            repository
                .Setup(x => x.Search(searchText))
                .Returns(new List<Domain.Todo.Search.Models.Todo>().AsReadOnly());
            
            var result = service.Search(searchText);

            result.ShouldBeOfType<ReadOnlyCollection<Domain.Todo.Search.Models.Todo>>();
        }
    }
}